nacos LDAP SSL握手问题

在使用nacos连接LDAP过程中，发现nacos链接LDAPS报错，如下：

`2022-10-21 16:29:12,456 ERROR [LDAP-LOGIN] failed

org.springframework.ldap.CommunicationException: 172.25.198.212:3269; nested exception is javax.naming.CommunicationException: 172.25.198.212:3269 [Root exception is javax.net.ssl.SSLHandshakeException: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target]
        at org.springframework.ldap.support.LdapUtils.convertLdapException(LdapUtils.java:108)
        at org.springframework.ldap.core.support.AbstractContextSource.createContext(AbstractContextSource.java:363)
        at org.springframework.ldap.core.support.AbstractContextSource.doGetContext(AbstractContextSource.java:147)
        at org.springframework.ldap.core.support.AbstractContextSource.getReadOnlyContext(AbstractContextSource.java:166)
        at org.springframework.ldap.core.LdapTemplate.search(LdapTemplate.java:361)
        at org.springframework.ldap.core.LdapTemplate.search(LdapTemplate.java:313)
        at org.springframework.ldap.core.LdapTemplate.search(LdapTemplate.java:646)
        at org.springframework.ldap.core.LdapTemplate.search(LdapTemplate.java:582)
        at org.springframework.ldap.core.LdapTemplate.authenticate(LdapTemplate.java:1445)
        at org.springframework.ldap.core.LdapTemplate.authenticate(LdapTemplate.java:1430)
        at org.springframework.ldap.core.LdapTemplate.authenticate(LdapTemplate.java:1373)
        at com.alibaba.nacos.plugin.auth.impl.LdapAuthenticationProvider.ldapLogin(LdapAuthenticationProvider.java:116)
        at com.alibaba.nacos.plugin.auth.impl.LdapAuthenticationProvider.authenticate(LdapAuthenticationProvider.java:80)
        at org.springframework.security.authentication.ProviderManager.authenticate(ProviderManager.java:182)
        at org.springframework.security.authentication.ProviderManager.authenticate(ProviderManager.java:201)
        at org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter$AuthenticationManagerDelegator.authenticate(WebSecurityConfigurerAdapter.java:518)
        at com.alibaba.nacos.plugin.auth.impl.NacosAuthManager.resolveTokenFromUser(NacosAuthManager.java:139)
        at com.alibaba.nacos.plugin.auth.impl.NacosAuthManager.resolveToken(NacosAuthManager.java:113)
        at com.alibaba.nacos.plugin.auth.impl.NacosAuthManager.login(NacosAuthManager.java:69)
        at com.alibaba.nacos.plugin.auth.impl.controller.UserController.login(UserController.java:224)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:205)
        at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:150)
        at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:117)
        at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:895)
        at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:808)
        at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87)
        at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1067)
        at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:963)
        at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1006)
        at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:909)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:681)
        at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:883)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:764)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:227)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
        at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
        at com.alibaba.nacos.console.filter.XssFilter.doFilterInternal(XssFilter.java:42)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
        at org.springframework.web.filter.CorsFilter.doFilterInternal(CorsFilter.java:91)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
        at com.alibaba.nacos.core.auth.AuthFilter.doFilter(AuthFilter.java:69)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
        at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:204)
        at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:183)
        at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:354)
        at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:267)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
        at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
        at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
        at org.springframework.boot.actuate.metrics.web.servlet.WebMvcMetricsFilter.doFilterInternal(WebMvcMetricsFilter.java:96)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
        at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:197)
        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97)
        at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:541)
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:135)
        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78)
        at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:687)
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:360)
        at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:399)
        at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
        at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:890)
        at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1743)
        at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
        at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
        at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
        at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
        at java.lang.Thread.run(Thread.java:745)
Caused by: javax.naming.CommunicationException: 172.25.198.212:3269
        at com.sun.jndi.ldap.Connection.<init>(Connection.java:226)
        at com.sun.jndi.ldap.LdapClient.<init>(LdapClient.java:137)
        at com.sun.jndi.ldap.LdapClient.getInstance(LdapClient.java:1614)
        at com.sun.jndi.ldap.LdapCtx.connect(LdapCtx.java:2746)
        at com.sun.jndi.ldap.LdapCtx.<init>(LdapCtx.java:319)
        at com.sun.jndi.ldap.LdapCtxFactory.getUsingURL(LdapCtxFactory.java:192)
        at com.sun.jndi.ldap.LdapCtxFactory.getUsingURLs(LdapCtxFactory.java:210)
        at com.sun.jndi.ldap.LdapCtxFactory.getLdapCtxInstance(LdapCtxFactory.java:153)
        at com.sun.jndi.ldap.LdapCtxFactory.getInitialContext(LdapCtxFactory.java:83)
        at javax.naming.spi.NamingManager.getInitialContext(NamingManager.java:684)
        at javax.naming.InitialContext.getDefaultInitCtx(InitialContext.java:313)
        at javax.naming.InitialContext.init(InitialContext.java:244)
        at javax.naming.ldap.InitialLdapContext.<init>(InitialLdapContext.java:154)
        at org.springframework.ldap.core.support.LdapContextSource.getDirContextInstance(LdapContextSource.java:42)
        at org.springframework.ldap.core.support.AbstractContextSource.createContext(AbstractContextSource.java:351)
        ... 90 common frames omitted
Caused by: javax.net.ssl.SSLHandshakeException: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
        at sun.security.ssl.Alerts.getSSLException(Alerts.java:192)
        at sun.security.ssl.SSLSocketImpl.fatal(SSLSocketImpl.java:1949)
        at sun.security.ssl.Handshaker.fatalSE(Handshaker.java:302)
        at sun.security.ssl.Handshaker.fatalSE(Handshaker.java:296)
        at sun.security.ssl.ClientHandshaker.serverCertificate(ClientHandshaker.java:1509)
        at sun.security.ssl.ClientHandshaker.processMessage(ClientHandshaker.java:216)
        at sun.security.ssl.Handshaker.processLoop(Handshaker.java:979)
        at sun.security.ssl.Handshaker.process_record(Handshaker.java:914)
        at sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:1062)
        at sun.security.ssl.SSLSocketImpl.performInitialHandshake(SSLSocketImpl.java:1375)
        at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1403)
        at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1387)
        at com.sun.jndi.ldap.Connection.createSocket(Connection.java:376)
        at com.sun.jndi.ldap.Connection.<init>(Connection.java:203)
        ... 104 common frames omitted
Caused by: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
        at sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:387)
        at sun.security.validator.PKIXValidator.engineValidate(PKIXValidator.java:292)
        at sun.security.validator.Validator.validate(Validator.java:260)
        at sun.security.ssl.X509TrustManagerImpl.validate(X509TrustManagerImpl.java:324)
        at sun.security.ssl.X509TrustManagerImpl.checkTrusted(X509TrustManagerImpl.java:229)
        at sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(X509TrustManagerImpl.java:124)
        at sun.security.ssl.ClientHandshaker.serverCertificate(ClientHandshaker.java:1491)
        ... 113 common frames omitted
Caused by: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
        at sun.security.provider.certpath.SunCertPathBuilder.build(SunCertPathBuilder.java:141)
        at sun.security.provider.certpath.SunCertPathBuilder.engineBuild(SunCertPathBuilder.java:126)
        at java.security.cert.CertPathBuilder.build(CertPathBuilder.java:280)
        at sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:382)
        ... 119 common frames omitted`


os: centos 
nacos version: 2.1.2

--- a/plugin-default-impl/src/main/java/com/alibaba/nacos/plugin/auth/impl/LdapAuthConfig.java
+++ b/plugin-default-impl/src/main/java/com/alibaba/nacos/plugin/auth/impl/LdapAuthConfig.java
@@ -29,11 +29,9 @@ import org.springframework.context.annotation.Configuration;
 import org.springframework.ldap.core.LdapTemplate;
 import org.springframework.ldap.core.support.LdapContextSource;
 
-import java.util.HashMap;
-import java.util.Map;
-
 /**
  * ldap auth config.
+ *
  * @author onewe
  */
 @Configuration
@@ -60,20 +58,13 @@ public class LdapAuthConfig {
     
     @Bean
     @Conditional(ConditionOnLdapAuth.class)
-    public LdapTemplate ldapTemplate() {
-        LdapContextSource contextSource = new LdapContextSource();
-        final Map<String, Object> config = new HashMap<>(16);
-        contextSource.setUrl(ldapUrl);
-        contextSource.setBase(ldapBaseDc);
-        contextSource.setUserDn(userDn);
-        contextSource.setPassword(password);
-        config.put("java.naming.ldap.attributes.binary", "objectGUID");
-        config.put("com.sun.jndi.ldap.connect.timeout", ldapTimeOut);
-        contextSource.setPooled(true);
-        contextSource.setBaseEnvironmentProperties(config);
-        contextSource.afterPropertiesSet();
-        return new LdapTemplate(contextSource);
-        
+    public LdapTemplate ldapTemplate(LdapContextSource ldapContextSource) {
+        return new LdapTemplate(ldapContextSource);
+    }
+    
+    @Bean
+    public LdapContextSource ldapContextSource() {
+        return new NacosLdapContextSource(ldapUrl, ldapBaseDc, userDn, password, ldapTimeOut);
     }
     
     @Bean

--- a/plugin-default-impl/src/main/java/com/alibaba/nacos/plugin/auth/impl/NacosLdapContextSource.java
+++ b/plugin-default-impl/src/main/java/com/alibaba/nacos/plugin/auth/impl/NacosLdapContextSource.java
@@ -1,0 +1,129 @@
+/*
+ * Copyright 1999-2018 Alibaba Group Holding Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.nacos.plugin.auth.impl;
+
+import com.alibaba.nacos.common.tls.TlsHelper;
+import org.springframework.ldap.core.support.LdapContextSource;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.Socket;
+import java.net.UnknownHostException;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.util.HashMap;
+import java.util.Map;
+import javax.naming.Context;
+import javax.net.ssl.SSLSocketFactory;
+
+/**
+ * NacosLdapContextSource.
+ *
+ * @author karsonto
+ */
+public class NacosLdapContextSource extends LdapContextSource {
+    
+    private final Map<String, Object> config = new HashMap<>(16);
+    
+    private boolean useTsl = false;
+    
+    private static final String LDAPS = "ldaps";
+    
+    public NacosLdapContextSource(String ldapUrl, String ldapBaseDc, String userDn, String password,
+            String ldapTimeOut) {
+        this.setUrl(ldapUrl);
+        this.setBase(ldapBaseDc);
+        this.setUserDn(userDn);
+        this.setPassword(password);
+        if (ldapUrl.toLowerCase().startsWith(LDAPS)) {
+            useTsl = true;
+        }
+        this.setPooled(true);
+        init(ldapTimeOut);
+    }
+    
+    /**
+     * init LdapContextSource config.
+     *
+     * @param ldapTimeOut ldap connection time out.
+     */
+    public void init(String ldapTimeOut) {
+        if (useTsl) {
+            System.setProperty("com.sun.jndi.ldap.object.disableEndpointIdentification", "true");
+            config.put("java.naming.security.protocol", "ssl");
+            config.put("java.naming.ldap.factory.socket", LdapSsLSocketFactory.class.getName());
+            config.put(Context.INITIAL_CONTEXT_FACTORY, "com.sun.jndi.ldap.LdapCtxFactory");
+        }
+        config.put("java.naming.ldap.attributes.binary", "objectGUID");
+        config.put("com.sun.jndi.ldap.connect.timeout", ldapTimeOut);
+        this.setBaseEnvironmentProperties(config);
+        this.afterPropertiesSet();
+    }
+    
+    @SuppressWarnings("checkstyle:EmptyLineSeparator")
+    public static class LdapSsLSocketFactory extends SSLSocketFactory {
+        
+        private SSLSocketFactory socketFactory;
+        
+        public LdapSsLSocketFactory() {
+            try {
+                this.socketFactory = TlsHelper.buildSslContext(true).getSocketFactory();
+            } catch (NoSuchAlgorithmException | KeyManagementException e) {
+                e.printStackTrace();
+            }
+        }
+        
+        @Override
+        public String[] getDefaultCipherSuites() {
+            return socketFactory.getDefaultCipherSuites();
+        }
+        
+        @Override
+        public String[] getSupportedCipherSuites() {
+            return socketFactory.getSupportedCipherSuites();
+        }
+        
+        @Override
+        public Socket createSocket(Socket socket, String s, int i, boolean b) throws IOException {
+            return socketFactory.createSocket(socket, s, i, b);
+        }
+        
+        @Override
+        public Socket createSocket(String s, int i) throws IOException, UnknownHostException {
+            return socketFactory.createSocket(s, i);
+        }
+        
+        @Override
+        public Socket createSocket(String s, int i, InetAddress inetAddress, int i1)
+                throws IOException, UnknownHostException {
+            return socketFactory.createSocket(s, i, inetAddress, i1);
+        }
+        
+        @Override
+        public Socket createSocket(InetAddress inetAddress, int i) throws IOException {
+            return socketFactory.createSocket(inetAddress, i);
+        }
+        
+        @Override
+        public Socket createSocket(InetAddress inetAddress, int i, InetAddress inetAddress1, int i1)
+                throws IOException {
+            return socketFactory.createSocket(inetAddress, i, inetAddress1, i1);
+        }
+    }
+    
+    
+}

--- a/plugin-default-impl/src/main/java/com/alibaba/nacos/plugin/auth/impl/NacosLdapContextSource.java
+++ b/plugin-default-impl/src/main/java/com/alibaba/nacos/plugin/auth/impl/NacosLdapContextSource.java
@@ -65,7 +65,7 @@ public class NacosLdapContextSource extends LdapContextSource {
         if (useTsl) {
             System.setProperty("com.sun.jndi.ldap.object.disableEndpointIdentification", "true");
             config.put("java.naming.security.protocol", "ssl");
-            config.put("java.naming.ldap.factory.socket", LdapSsLSocketFactory.class.getName());
+            config.put("java.naming.ldap.factory.socket", LdapSslSocketFactory.class.getName());
             config.put(Context.INITIAL_CONTEXT_FACTORY, "com.sun.jndi.ldap.LdapCtxFactory");
         }
         config.put("java.naming.ldap.attributes.binary", "objectGUID");
@@ -75,11 +75,11 @@ public class NacosLdapContextSource extends LdapContextSource {
     }
     
     @SuppressWarnings("checkstyle:EmptyLineSeparator")
-    public static class LdapSsLSocketFactory extends SSLSocketFactory {
+    public static class LdapSslSocketFactory extends SSLSocketFactory {
         
         private SSLSocketFactory socketFactory;
         
-        public LdapSsLSocketFactory() {
+        public LdapSslSocketFactory() {
             try {
                 this.socketFactory = TlsHelper.buildSslContext(true).getSocketFactory();
             } catch (NoSuchAlgorithmException | KeyManagementException e) {

--- a/plugin-default-impl/src/main/java/com/alibaba/nacos/plugin/auth/impl/NacosLdapContextSource.java
+++ b/plugin-default-impl/src/main/java/com/alibaba/nacos/plugin/auth/impl/NacosLdapContextSource.java
@@ -17,6 +17,7 @@
 package com.alibaba.nacos.plugin.auth.impl;
 
 import com.alibaba.nacos.common.tls.TlsHelper;
+import com.alibaba.nacos.core.utils.Loggers;
 import org.springframework.ldap.core.support.LdapContextSource;
 
 import java.io.IOException;
@@ -83,7 +84,7 @@ public class NacosLdapContextSource extends LdapContextSource {
             try {
                 this.socketFactory = TlsHelper.buildSslContext(true).getSocketFactory();
             } catch (NoSuchAlgorithmException | KeyManagementException e) {
-                e.printStackTrace();
+                Loggers.AUTH.error("Failed to create SSLContext", e);
             }
         }
         

--- a/test/core-test/src/test/java/com/alibaba/nacos/test/core/auth/LdapAuth_ITCase.java
+++ b/test/core-test/src/test/java/com/alibaba/nacos/test/core/auth/LdapAuth_ITCase.java
@@ -21,7 +21,9 @@ import com.alibaba.nacos.auth.config.AuthConfigs;
 import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
+import org.junit.runners.Suite;
 import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
 import org.springframework.boot.test.context.SpringBootTest;
 import org.springframework.boot.test.mock.mockito.MockBean;
 import org.springframework.boot.web.server.LocalServerPort;
@@ -31,7 +33,8 @@ import org.springframework.test.context.junit4.SpringRunner;
 import java.net.URL;
 import java.util.concurrent.TimeUnit;
 
-
+@RunWith(Suite.class)
+@Suite.SuiteClasses({ LdapAuth_ITCase.NonTlsTest.class,LdapAuth_ITCase.TlsTest.class})
 public class LdapAuth_ITCase extends AuthBase {
     
     @LocalServerPort
@@ -66,7 +69,8 @@ public class LdapAuth_ITCase extends AuthBase {
     
     @RunWith(SpringRunner.class)
     @SpringBootTest(classes = Nacos.class, properties = {"server.servlet.context-path=/nacos",
-            "nacos.core.auth.system.type=ldap","nacos.core.auth.ldap.url=ldaps://localhost:636"}, webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+            "nacos.core.auth.system.type=ldap",
+            "nacos.core.auth.ldap.url=ldaps://localhost:636"}, webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
     public static class TlsTest extends LdapAuth_ITCase {
         
         @Test

--- a/test/core-test/src/test/java/com/alibaba/nacos/test/core/auth/LdapAuth_ITCase.java
+++ b/test/core-test/src/test/java/com/alibaba/nacos/test/core/auth/LdapAuth_ITCase.java
@@ -17,6 +17,7 @@
 package com.alibaba.nacos.test.core.auth;
 
 import com.alibaba.nacos.Nacos;
+import com.alibaba.nacos.auth.config.AuthConfigs;
 import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
@@ -27,9 +28,10 @@ import org.springframework.boot.web.server.LocalServerPort;
 import org.springframework.ldap.core.LdapTemplate;
 import org.springframework.test.context.junit4.SpringRunner;
 
-@RunWith(SpringRunner.class)
-@SpringBootTest(classes = Nacos.class, properties = {"server.servlet.context-path=/nacos",
-        "nacos.core.auth.system.type=ldap"}, webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+import java.net.URL;
+import java.util.concurrent.TimeUnit;
+
+
 public class LdapAuth_ITCase extends AuthBase {
     
     @LocalServerPort
@@ -44,11 +46,34 @@ public class LdapAuth_ITCase extends AuthBase {
     public void init() throws Exception {
         Mockito.when(ldapTemplate.authenticate("", "(" + filterPrefix + "=" + "karson" + ")", "karson"))
                 .thenReturn(true);
+        AuthConfigs.setCachingEnabled(false);
+        TimeUnit.SECONDS.sleep(5L);
+        String url = String.format("http://localhost:%d/", port);
+        System.setProperty("nacos.core.auth.enabled", "true");
+        this.base = new URL(url);
     }
     
-    @Test
-    public void testLdapAuth() throws Exception {
-        super.login("karson", "karson");
+    @RunWith(SpringRunner.class)
+    @SpringBootTest(classes = Nacos.class, properties = {"server.servlet.context-path=/nacos",
+            "nacos.core.auth.system.type=ldap"}, webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+    public static class NonTlsTest extends LdapAuth_ITCase {
+        
+        @Test
+        public void testLdapAuth() throws Exception {
+            super.login("karson", "karson");
+        }
     }
+    
+    @RunWith(SpringRunner.class)
+    @SpringBootTest(classes = Nacos.class, properties = {"server.servlet.context-path=/nacos",
+            "nacos.core.auth.system.type=ldap","nacos.core.auth.ldap.url=ldaps://localhost:636"}, webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+    public static class TlsTest extends LdapAuth_ITCase {
+        
+        @Test
+        public void testLdapAuth() throws Exception {
+            super.login("karson", "karson");
+        }
+    }
+    
     
 }

--- a/test/core-test/src/test/java/com/alibaba/nacos/test/core/auth/LdapAuth_ITCase.java
+++ b/test/core-test/src/test/java/com/alibaba/nacos/test/core/auth/LdapAuth_ITCase.java
@@ -23,7 +23,6 @@ import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.Suite;
 import org.mockito.Mockito;
-import org.mockito.junit.MockitoJUnitRunner;
 import org.springframework.boot.test.context.SpringBootTest;
 import org.springframework.boot.test.mock.mockito.MockBean;
 import org.springframework.boot.web.server.LocalServerPort;

